tools/update_vcpkg.py

```diff
@@ -0,0 +1,168 @@
+import re
+import sys
+
+from datetime import datetime
+from urllib.request import urlopen
+from urllib.error import HTTPError
+
+MIXXX_DOWNLOAD_BASE = "https://downloads.mixxx.org/dependencies/"
+DEFAULT_PATTERN = (
+    r'BUILDENV_NAME="(mixxx-deps-[0-9]\.[0-9]-{triplet}-)'
+    r'([a-z0-9]+)"\n(\W+)BUILDENV_SHA256="([^"]+)"'
+)
+DEFAULT_REPLACE = (
+    r'BUILDENV_NAME="\g<1>{version}"\n\g<3>BUILDENV_SHA256="{shasum}"'
+)
+PLATFORMS = [
+    {
+        "os": "Windows",
+        "triplet": "arm64-windows",
+        "release_triplet": "arm64-windows-rel",
+        "file": "tools/windows_buildenv.bat",
+        "pattern": (
+            r"BUILDENV_NAME=(mixxx-deps-[0-9]\.[0-9]-{triplet}-)"
+            r"([a-z0-9]+)\n(\W+)SET BUILDENV_SHA256=([^\W]+)"
+        ),
+        "replace": (
+            r"BUILDENV_NAME=\g<1>{version}\n\g<3>"
+            r"SET BUILDENV_SHA256={shasum}"
+        ),
+    },
+    {
+        "os": "Windows",
+        "triplet": "x64-windows",
+        "release_triplet": "x64-windows-rel",
+        "file": "tools/windows_buildenv.bat",
+        "pattern": (
+            r"BUILDENV_NAME=(mixxx-deps-[0-9]\.[0-9]-{triplet}-)"
+            r"([a-z0-9]+)\n(\W+)SET BUILDENV_SHA256=([^\W]+)"
+        ),
+        "replace": (
+            r"BUILDENV_NAME=\g<1>{version}\n\g<3>"
+            r"SET BUILDENV_SHA256={shasum}"
+        ),
+    },
+    {
+        "os": "macOS",
+        "triplet": "arm64-osx-cross",
+        "release_triplet": "arm64-osx-cross-rel",
+        "file": "tools/macos_buildenv.sh",
+    },
+    {
+        "os": "macOS",
+        "triplet": "x64-osx",
+        "release_triplet": "x64-osx-rel",
+        "file": "tools/macos_buildenv.sh",
+    },
+    {
+        "os": "macOS",
+        "triplet": "arm64-osx",
+        "release_triplet": "arm64-osx-rel",
+        "file": "tools/macos_buildenv.sh",
+    },
+    {
+        "os": "Linux",
+        "triplet": "arm64-android",
+        "release_triplet": "arm64-android-release",
+        "file": "tools/android_buildenv.sh",
+    },
+]
+
+
+def update(defs, channel, data):
+    with open(defs["file"], "r") as file:
+        content = file.read()
+
+    # Perform the replacement
+    pattern = defs.get("pattern", DEFAULT_PATTERN).format(**data)
```

Comment:
Here a check would be nice if the version already exists or of the file has been written.